### PR TITLE
feat(Breadcrumb): review

### PR DESCRIPTION
**Pull request by marielaures**
Implementing the Breadcrumb component.
You can see the issue here: https://github.com/algolia/react-instantsearch/issues/23
and check how it works in Storybook!

### CODE REVIEW

File: packages/react-instantsearch-theme-algolia/style.scss [hunk 1]
```diff
@@ -3,111 +3,105 @@
...
+        transform: translateY(-50%)
+    }
 }
 
 @mixin openCaret() {
-  &:after {
-    top: 25%;
-    transform: rotate(90deg)
-  }
+    &:after {
```
Comment (Haroenv) [inline packages/react-instantsearch-theme-algolia/style.scss]: I'm not sure if these style changes were on purpose. Could you do it in another PR if they're on purpose so that the diff is more readable?

File: packages/react-instantsearch/src/components/Breadcrumb.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    label: PropTypes.string.isRequired,
+    value: PropTypes.string.isRequired,
+  })
+);
+
+class Breadcrumb extends Component {
+  static propTypes = {
+    canRefine: PropTypes.bool.isRequired,
+    createURL: PropTypes.func.isRequired,
+    items: itemsPropType,
```
Comment (Haroenv) [inline packages/react-instantsearch/src/components/Breadcrumb.js]: Not a proptypes expert, but this should be required too right, since there needs to be an array, with at least one item.

File: packages/react-instantsearch/src/components/Breadcrumb.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    refine: PropTypes.func.isRequired,
+    rootURL: PropTypes.string,
+    separator: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
```
Comment (Haroenv) [inline packages/react-instantsearch/src/components/Breadcrumb.js]: With react 16, a string is an element too, does that matter here?

File: packages/react-instantsearch/src/components/Breadcrumb.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+          onClick={() => (!this.props.rootURL ? refine() : null)}
+          href={this.props.rootURL ? this.props.rootURL : createURL()}
+        >
+          <span {...cx('rootLabel')}>
+            {translate('rootLabel')}
+          </span>
+          <span {...cx('separator')}>
+            {this.props.separator}
+          </span>
+
```
Comment (Haroenv) [inline packages/react-instantsearch/src/components/Breadcrumb.js]: nit: I don't think this whitespace is needed

File: packages/react-instantsearch/src/components/__snapshots__/Breadcrumb.test.js.snap [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    <span
+      className="ais-Breadcrumb__rootLabel"
+    >
+      Home
+    </span>
+    <span
+      className="ais-Breadcrumb__separator"
+    >
+      <span>
+        🔍
```
Comment (Haroenv) [inline packages/react-instantsearch/src/components/__snapshots__/Breadcrumb.test.js.snap]: The separator should be in-between the two links, no?

File: packages/react-instantsearch/src/connectors/hierarchicalMenuLogic.js [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+import { SearchParameters } from 'algoliasearch-helper';
+import {
+  cleanUpValue,
+  getCurrentRefinementValue,
+  getIndex,
+  getResults,
+  refineValue,
+} from '../core/indexUtils';
+
+export const getId = props => props.attributes[0];
```
Comment (Haroenv) [inline packages/react-instantsearch/src/connectors/hierarchicalMenuLogic.js]: I don't really understand this file, can it be reused in `connectHierarchicalMenu` or is that a stupid idea?

Comment (mthuret) [inline packages/react-instantsearch/src/connectors/hierarchicalMenuLogic.js]: This file needs to be removed (was used before, then refactored).

File: packages/react-instantsearch/src/components/Breadcrumb.js (hunk 2 above)
Comment (marielaures) [inline packages/react-instantsearch/src/components/Breadcrumb.js]: So, I was asking myself the same question, but I checked on the HierarchicalMenu and on the List components, and apparently their `itemsPropType` isn't required either?

File: packages/react-instantsearch/src/components/Breadcrumb.test.js [hunk 7]
```diff
@@ -60,11 +60,11 @@ describe('Breadcrumb', () => {
     );
 
     const items = wrapper.find('.ais-Breadcrumb__itemLink');
-    expect(items.length).toBe(4); //Home + Breadcrumb (3 levels)
+    expect(items.length).toBe(4);
 
     items.first().simulate('click');
     expect(refine.mock.calls.length).toBe(1);
-    expect(refine.mock.calls[0][0]).toEqual(); //Home = empty refinement
+    expect(refine.mock.calls[0][0]).toEqual();
```
Comment (Haroenv) [inline packages/react-instantsearch/src/components/Breadcrumb.test.js]: I don't understand why you're calling `toEqual()` without any arguments. What about using [`.toBeNull()`](http://facebook.github.io/jest/docs/expect.html#tobenull)

File: packages/react-instantsearch/src/components/Breadcrumb.js (hunk 3 above)
Comment (marielaures) [inline packages/react-instantsearch/src/components/Breadcrumb.js]: @Haroenv Do you mean there is no need mention `PropTypes.string`?

Comment (Haroenv) [inline packages/react-instantsearch/src/components/Breadcrumb.js]: That's what I mean, but for consistency, let's allow both until we don't have react 15 users anymore.

File: packages/react-instantsearch/src/components/__snapshots__/Breadcrumb.test.js.snap [hunk 8]
```diff
@@ -0,0 +1,223 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Breadcrumb has a separator prop that can be a custom component 1`] = `
+<div
+  className="ais-Breadcrumb__root"
+>
+  <div
```
Comment (Haroenv) [inline packages/react-instantsearch/src/components/__snapshots__/Breadcrumb.test.js.snap]: I think that a breadcrumb item should be a span, so that the default rendering without css is like what we expect: 

div|span
---|---
screen shot 2017-08-03 at 15 45 06 | screen shot 2017-08-03 at 15 45 00

File: packages/react-instantsearch/src/connectors/connectBreadcrumb.js [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+    }
+    return acc;
+  }, []);
+}
+
+/**
+ * The breadcrumb component is s a type of secondary navigation scheme that 
+ * reveals the user’s location in a website or web application.
+ * 
+ * If you want to use the "defaultRefinement" prop, you can do so by using a Virtual Hierarchical Menu
```
Comment (mthuret) [inline packages/react-instantsearch/src/connectors/connectBreadcrumb.js]: I think this should be rephrase a bit. 
The breadcrumb doesn't have any `defaultRefinement` prop, so this might be confusing. 

Maybe have something like => "If you want to select a specific refinement for your Breadcrumb you'll need to use a Virtual Hierarchical Menu..." And then present a usage example. 

This should be present also on the widget guide and maybe only on this one.
